### PR TITLE
add tests for stream standard

### PR DESCRIPTION
**Pull request by anonrig**
Tests certain edge cases which are not covered in standard.c++

### CODE REVIEW

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 1]
```diff
@@ -0,0 +1,1375 @@
...
+        }
+      },
+    });
+
+    const reader = rs.getReader();
+
+    // First read gets the data
+    const result1 = await reader.read();
+    strictEqual(result1.value, 'data');
+    strictEqual(result1.done, false);
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: It should be documented that this is actually incorrect behavior that we should likely fix. The expectation in this case (as evidenced in other runtimes) is that the first read should reject with an error. The read should NOT actually be fulfilled.

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 2]
```diff
@@ -0,0 +1,1375 @@
...
+    // Start a read to trigger pull which sets pending close state
+    const readPromise = reader.read();
+
+    // Release reader so we can cancel
+    reader.releaseLock();
+
+    // Cancel should handle pending state
+    await rs.cancel('cancel reason');
+
+    // The read should complete (with done=true due to close)
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: Per the spec, the read should be canceled when the lock is released. This is another deviation from the spec we should fix.

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 3]
```diff
@@ -0,0 +1,1375 @@
...
+        controller.close();
+      },
+    });
+
+    const reader = rs.getReader({ mode: 'byob' });
+    const result = await reader.read(new Uint8Array(10));
+
+    strictEqual(result.done, true);
+    ok(result.value instanceof Uint8Array);
+    strictEqual(result.value.byteLength, 0);
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js:284]: This should likely also verify that it's the same backing store as the input.

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 4]
```diff
@@ -0,0 +1,1375 @@
...
+export const writableStreamBackpressure = {
+  async test() {
+    let resolveWrite;
+    const ws = new WritableStream(
+      {
+        write() {
+          return new Promise((resolve) => {
+            resolveWrite = resolve;
+          });
+        },
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js:440]:
```suggestion
        write() {
          const { promise, resolve } = Promise.withResolvers();
          resolveWrite = resolve;
          return promise;
        },
```

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 5]
```diff
@@ -0,0 +1,1375 @@
...
+    strictEqual(writer.desiredSize, 1);
+    const writePromise = writer.write('a');
+    strictEqual(writer.desiredSize, 0);
+    const writePromise2 = writer.write('b');
+    strictEqual(writer.desiredSize, -1);
+
+    resolveWrite();
+    await writePromise;
+    resolveWrite();
+    await writePromise2;
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js:456]: Should check that `desiredSize` returns back to 1

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 6]
```diff
@@ -0,0 +1,1375 @@
...
+  async test() {
+    let controller;
+    const rs = new ReadableStream(
+      {
+        type: 'bytes',
+        start(c) {
+          controller = c;
+        },
+        pull(c) {
+          // Respond with less data than requested to keep BYOB request alive
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: What does the "keep BYOB request alive" part here mean? Calling `respond(1)` fulfills the request. There's nothing "kept alive"

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 7]
```diff
@@ -0,0 +1,1375 @@
...
+    const reader = rs.getReader({ mode: 'byob' });
+    const result = await reader.read(new Uint8Array(10));
+
+    strictEqual(result.value[0], 42);
+    strictEqual(result.value.byteLength, 1);
+  },
+};
+
+// Test consumerCount edge case (lines 954-967)
+// This is triggered when checking consumer count during certain operations
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: can you describe the "certain operations" more here. It's not clear exactly what edge case this is testing.

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 8]
```diff
@@ -0,0 +1,1375 @@
...
+export const cancelClosedStream = {
+  async test() {
+    const rs = new ReadableStream({
+      start(controller) {
+        controller.close();
+      },
+    });
+
+    // Cancel should resolve immediately for closed stream
+    await rs.cancel();
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js:611]: It's not clear what exactly this is testing. There are no assertions and there's no indication of what the failure case would be.

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 9]
```diff
@@ -0,0 +1,1375 @@
...
+
+// Test Response.text() with stream that returns empty chunks
+export const responseTextWithEmptyChunks = {
+  async test() {
+    const rs = new ReadableStream({
+      start(c) {
+        c.enqueue(new TextEncoder().encode('hello'));
+        c.enqueue(new Uint8Array(0)); // Empty chunk - should be skipped
+        c.enqueue(new TextEncoder().encode(' world'));
+        c.close();
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: There's really no way of verifying in this test that the empty chunk was actually skipped.

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 10]
```diff
@@ -0,0 +1,1375 @@
...
+    strictEqual(text, 'hello world');
+  },
+};
+
+// Test Response.arrayBuffer() with stream that returns non-bytes
+export const responseArrayBufferWithNonBytes = {
+  async test() {
+    // Create a stream that returns a non-byte value
+    // This is tricky because workerd streams are typically byte-oriented
+    // We need to use internal APIs or specific conditions
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: This comment does not make sense. You're returning byte data. You're not creating "a stream that returns a non-byte value"

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 11]
```diff
@@ -0,0 +1,1375 @@
...
+        if (pullCount >= 3) {
+          c.close();
+        }
+      },
+    });
+
+    const reader = rs.getReader({ mode: 'byob' });
+
+    const r1 = await reader.read(new Uint8Array(5));
+    strictEqual(r1.value[0], 1);
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js:671]: These should be checking the lengths of `value[0]` also

File: src/workerd/api/tests/streams-standard-coverage-test.js [hunk 12]
```diff
@@ -0,0 +1,1375 @@
...
+  },
+};
+
+// Test byte stream without autoAllocateChunkSize using default reader
+export const byteStreamDefaultReaderNoAutoAllocate = {
+  async test() {
+    let pullCount = 0;
+    const rs = new ReadableStream({
+      type: 'bytes',
+      // No autoAllocateChunkSize - pull must use enqueue
```
Comment (jasnell) [inline src/workerd/api/tests/streams-standard-coverage-test.js]: Our implementation, by default, always sets autoAllocateChunkSize. There's a new compat flag to enable more standard behavior. Specifically, without that compat flag this comment is incorrect.

File: src/workerd/api/tests/streams-standard-coverage-test.js (hunk 6 above)
Comment (anonrig) [reply src/workerd/api/tests/streams-standard-coverage-test.js]: Fixing it.

File: src/workerd/api/tests/streams-standard-coverage-test.js (hunk 8 above)
Comment (anonrig) [reply src/workerd/api/tests/streams-standard-coverage-test.js:611]: This got removed due to a local git conflict. Fixing it now.

**Comment by anonrig** [comment]
Closing this pull-request since it doesn't reach to edge cases where I was planning on testing.
